scripts/events_module/short/handle_short_events.py

```diff
@@ -133,56 +133,53 @@ def handle_event(
             event_type = "death"
         elif event_type == "health":
             event_type = "injury"
-        possible_short_events = GenerateEvents.possible_short_events(event_type)
 
-        final_events = GenerateEvents.filter_possible_short_events(
-            Cat_class=Cat,
-            possible_events=possible_short_events,
-            cat=self.main_cat,
-            random_cat=self.random_cat,
-            other_clan=self.other_clan,
-            freshkill_active=FRESHKILL_EVENT_ACTIVE,
-            freshkill_trigger_factor=FRESHKILL_EVENT_TRIGGER_FACTOR,
-            sub_types=self.sub_types,
-            allowed_events=self.allowed_events,
-            excluded_events=self.excluded_events,
-            ignore_subtyping=ignore_subtyping,
-        )
+        # choosing frequency
+        # think of it as "in a span of 10 moons, in how many moons should this sort of event appear?"
+        frequency_roll = randint(1, 10)
+        if frequency_roll <= 4:
+            weight = 4
+        elif frequency_roll <= 7:
+            weight = 3
+        elif frequency_roll <= 9:
+            weight = 2
+        else:
+            weight = 1
+
+        chosen_event = None
+        while not chosen_event and weight < 5:
+            possible_short_events = GenerateEvents.possible_short_events(
+                event_type, weight
+            )
+
+            chosen_event, random_cat = GenerateEvents.filter_possible_short_events(
+                Cat_class=Cat,
+                possible_events=possible_short_events,
+                cat=self.main_cat,
+                random_cat=self.random_cat,
+                other_clan=self.other_clan,
+                freshkill_active=FRESHKILL_EVENT_ACTIVE,
+                freshkill_trigger_factor=FRESHKILL_EVENT_TRIGGER_FACTOR,
+                sub_types=self.sub_types,
+                allowed_events=self.allowed_events,
+                excluded_events=self.excluded_events,
+                ignore_subtyping=ignore_subtyping,
+            )
+            if not chosen_event:
+                # we'll see if any higher weighted events are available
```

Comment:
```suggestion
                # we'll see if any more common events are available
```
just for my own brain lol